### ISSUE TITLE
Unable to select end date when using date range in a specific case

### ISSUE THREAD

**Issue by rahulgadhyan** [logbon72/angular-material-datetimepicker#130]
I am using start date and end date selection range with following condition:
1) end date cannot be less than start date

If the current system time is in AM, say 13/05/2018, 09:00 AM and i want to select date range as 12/05/2018-10:00PM to 12/05/2018-11:00PM.

After Selecting start date-time as 12/05/2018-10:00PM, i am not able to select end date-time as 12/05/2018-11:00PM because when I open the end date-time picker, it shown todays date, ie 13th (and time as 9:00 AM). and if I try to select 12th, it cannot select as 12/5/18, 9:00 AM is lesser than start date already selected.

I would like to know if there is a way to work around this particular isssue
Thanks

**Comment by hexadecy** [comment]
Can you paste some template code and why the end date-time is set to today?

**Comment by rahulgadhyan** [comment]
this is how the code is handled in html:


<ul>
    <li class="row date-size">
        <md-input-container flex-gt-md="90" class="dateInput date-size">
            <input mdc-datetime-picker=""
                   date="true" time="true" short-time="true"
                   type="text" id="startDate"
                   placeholder="{{ 'STARTTIME' | translate }}"
show-todays-date=""
                   ng-model="$parent.fromDate"
                   show-icon="true"
                   auto-ok="true"
                   max-date="maxStartDate"
                   ng-change="checkStartEndDate()"
                   class=" dtp-no-msclear dtp-input md-input">
        </md-input-container>
    </li>
    <li class="row">
        <md-input-container flex-gt-md="90" class="dateInput date-size">
            <input mdc-datetime-picker=""
                   date="true" time="true" short-time="true"
                   type="text" id="endDate"
                   placeholder="{{ 'ENDTIME' | translate }}" show-todays-date=""
                   ng-model="$parent.toDate"
                   show-icon="true"
                   auto-ok="true"
                   min-date="fromDate"
                   max-date="maxDate"
                   ng-change="setMaxStartDate()"
                   class=" dtp-no-msclear dtp-input md-input">
        </md-input-container>
    </li>

</ul>



And have the following code in javascript (using angular framework):

$scope.setMaxStartDate = function () {
    $scope.closeAlert();
    if($scope.fromDate == null)
        $scope.maxStartDate = $scope.toDate;
    $scope.checkStartEndDate();
}

$scope.checkStartEndDate = function () {
    $scope.closeAlert();
    if($scope.fromDate > $scope.toDate)
        $scope.toDate = null;
    //if($scope.toDate==null && current system time = AM &&
$scope.fromDate = yesterday PM) then set $scope.toDate as
$scope.fromDate
}

im using this code in an app. Now if current system time is say 17/5/18 -
9:00 AM.

I need to set the start and end date as follows:

start: 16/5/18, 10:00 PM

end: 16/5/18, 11:00 PM

I select the start date and enter as required.

Then I select the end date. currently it shows 17/5/18. If i select
17/5/18, i get the time auto set at 9:00AM, i.e. the current time.

This is the expected behaviour.

However, when i click the end date and select 16/5/18. I guesss, the
time still refers to 9:00 AM so, effectively the ref is to
16/5/18-9:00AM

Now as i have set that the end date has to be greater than start date,
i am not able to select anything as 16/5/18-9:00AM is lesser than the
already selected start date.

If I try to select PM, as the hour reading refers to 9, it will point
to 9:00PM which is still smaller than 10:00PM selected in the start
date.

So it does not allow me to select PM. I am completely stuck.


I hope i have been able to clear my question

Thanks for your time

Regards

Rahul

**Comment by hexadecy** [comment]
What version are you using? v1.15.1 ?

I tried to reproduce it with you code, but it's working as I think it should.
1. I set manually my system time to 17/5/18 - 9:00 AM
2. start: 16/5/18, 10:00 PM
3. end: 16/5/18, 11:00 PM
4. go to change end again, but I clicked "today"
5. Then change date to 16/5/18 and the time stay at 9:00 AM
6. Cannot change time, but I can click on PM circle
7. Then I can only select 10 PM or 11PM in hours. Max 11:59PM

Please see my video of this:
https://drive.google.com/file/d/1M-y5GL_SpNVzTdoaNmm_xuOiA866cMbh/view?usp=sharing

**Comment by hexadecy** [comment]
@rahulgadhyan have you found something?

**Comment by rahulgadhyan** [comment]
Hi Michel,

I was busy with some other portion of our code. I shall try to work as
directed this week. If the issue persists, I shall record the video and
share it with you

Thanks a lot for your time and effort

Regards
Rahul

**Comment by hexadecy** [comment]
Yes @rahulgadhyan  you can record a video with something like Screencastify.
And add many info as you can on how to reproduce it.

**Comment by hexadecy** [comment]
I fixed this in https://github.com/beenote/angular-material-datetimepicker/issues/54.
